src/core/qgscadutils.cpp

```diff
@@ -380,18 +384,38 @@ QgsCadUtils::AlignMapPointOutput QgsCadUtils::alignMapPoint( const QgsPointXY &o
       };
 
       const QgsFeature feature = snap.layer()->getFeature( snap.featureId() );
-      const QgsGeometry geom = feature.geometry();
+      const QgsGeometry geometry = feature.geometry();
+      const QgsAbstractGeometry *geom = geometry.constGet();
 
-      bool checked = checkLineExtension( geom.vertexAt( snap.vertexIndex() - 1 ) );
-      if ( checked )
+      QgsVertexId vertexId;
+      geometry.vertexIdFromVertexNr( snap.vertexIndex(), vertexId );
+      if ( vertexId.isValid() )
       {
-        res.softLockLineExtension = Qgis::LineExtensionSide::BeforeVertex;
-      }
+        QgsVertexId previousVertexId = QgsVertexId( vertexId.part, vertexId.ring, vertexId.vertex - 1 );
```

Comment:
This will have an issue if vertexId.vertex == 0

In any case, we should use:

```suggestion
        QgsVertexId previousVertexId;
        QgsVertexId nextVertexId;
        geom->adjacentVertices( vertexId, previousVertex, nextVertex );
```

which should already cover all the necessary logic

Reply:
@nyalldawson , done.